qemu/tests/numa_stress.py

```diff
@@ -68,89 +72,140 @@ def run(test, params, env):
-                    "skipping test...")
-
     if not os.path.isdir(tmpfs_path):
         os.mkdir(tmpfs_path)
 
     test_mem = float(params.get("mem"))*float(params.get("mem_ratio", 0.8))
-    stress_args = "--cpu 4 --io 4 --vm 2 --vm-bytes %sM" % int(test_mem / 2)
     most_used_node, memory_used = max_mem_map_node(host_numa_node, qemu_pid)
+    os_type = params["os_type"]
+
+    if os_type == "windows":
+        install_path = params["install_path"]
+        heavyload_install()
+    else:
+        stress_args = "--cpu %s --vm-bytes %sM --timeout %s" % (maxcpus,
+                                                                test_mem,
+                                                                stress_time)
+        stress_test = utils_test.VMStress(vm, "stress", params,
+                                          stress_args=stress_args)
+        stress_test.install()
 
     for test_round in range(test_count):
         if os.path.exists(memory_file):
             os.remove(memory_file)
         utils_memory.drop_caches()
-        if utils_memory.freememtotal() < tmpfs_size:
-            test.error("Don't have enough memory to execute this "
-                       "test after %s round" % test_round)
         error_context.context("Executing stress test round: %s" % test_round,
                               logging.info)
         numa_node_malloc = most_used_node
+        tmpfs_size = int(host_numa_node.read_from_node_meminfo(numa_node_malloc,
+                                                               "MemFree"))
+        if utils_memory.freememtotal() - tmpfs_size < 1024*1024:
+            test.cancel("Host does not have enough free memory to run the test,"
+                        "skipping test...")
+        dd_timeout = tmpfs_size / tmpfs_write_speed * 1.5
+        mount_fs_size = "size=%dK" % tmpfs_size
+        dd_cmd = "dd if=/dev/urandom of=%s bs=1k count=%s" % (memory_file,
+                                                              tmpfs_size)
         numa_dd_cmd = "numactl -m %s %s" % (numa_node_malloc, dd_cmd)
         error_context.context("Try to allocate memory in node %s"
                               % numa_node_malloc, logging.info)
         try:
             utils_misc.mount("none", tmpfs_path, "tmpfs", perm=mount_fs_size)
             funcatexit.register(env, params.get("type"), utils_misc.umount,
                                 "none", tmpfs_path, "tmpfs")
-            process.system(numa_dd_cmd, timeout=dd_timeout, shell=True)
-        except Exception as error_msg:
-            if "No space" in str(error_msg):
-                pass
+            try:
+                process.system(numa_dd_cmd, timeout=dd_timeout, shell=True)
+            except OSError as e:
+                if e.errno == errno.ENOSPC:
+                    pass
+                else:
+                    test.fail("Can not allocate memory in node %s."
+                              " Error message:%s" % (numa_node_malloc, str(e)))
```

Comment:
how about ?
```
               if e.errno != errno.ENOSPC:
                    test.fail("Can not allocate memory in node %s."
                              " Error message:%s" % (numa_node_malloc, str(e)))
```

Reply:
Thanks for your review. Will update later.